src/dune

```diff
@@ -1,16 +1,26 @@
 (library
   (public_name sqlite3)
-  (c_names sqlite3_stubs)
+  (c_names sqlite3_stubs sqlite3)
   (c_flags (
-    (:include c_flags.sexp) -g -O2 -fPIC -DPIC
+    (:include c_flags.sexp) -I .
     ; NOTE: for debugging before releases
     ; -Wall -pedantic -Wextra -Wunused -Wno-long-long -Wno-keyword-macro
   ))
-  (c_library_flags (:include c_library_flags.sexp) -lpthread)
+  (c_library_flags (:include c_library_flags.sexp))
 )
 
 (rule
   (targets c_flags.sexp c_library_flags.sexp)
   (deps (:discover config/discover.exe))
   (action (run %{discover}))
 )
+
+(rule
+  (targets sqlite3.c sqlite3.h)
+  (action
+    (progn
+      (system "curl -L https://www.sqlite.org/2019/sqlite-amalgamation-3270200.zip -o sqlite-amalgamation.zip")
```

Comment:
This won't work in `opam`. You cannot download things in the build phase.

Reply:
Probably when sandboxing is enabled? It does work for me in opam under Windows...
If sandboxing is really an issue, can you suggest a workaround?

Reply:
> If sandboxing is really an issue, can you suggest a workaround?

Well since you basically hard-coded everything I don't see exactly why you don't integrate the files in the repo itself, but I don't suggest this should be done.

Why don't we try to make this work with `pkg-config`, what are the problems exactly ?

Reply:
> Why don't we try to make this work with pkg-config, what are the problems exactly ?

As the title implies, this PR is aimed to support Windows build. You've got neither `pkg-config` no compiled sqlite3 library out-of-the box there (and no `depext` support in OPAM). Personally I think that the way to go is `depext` with [conan.io](https://conan.io) support (that provides C library binary packages on any platform) but right now building sqlite3 from sources seems to be the simplest solution.

Reply:
> As the title implies, this PR is aimed to support Windows build.

I don't have an opinion of what should be used on Windows but IIUC you simply blew away the pkg-config support for other platforms here which I don't find a very good idea.

If Windows users find your solution acceptable maybe you could play with the `extra-source` opam field (sadly it seems it cannot be filtered, so everyone will have to download it) and add a few windows specific `install:` opam steps.

See for example this [opam file](https://github.com/ocaml/opam-repository/blob/a4d2989b5c7c41c3f05a62856a3982c04e2c7086/packages/libtorch/libtorch.1.0.1/opam).

Reply:
or even make a `win-libsqlite3` package that builds and install sqlite on windows and depend on it only on windows, though I don't know what the stance of the opam-repository maintainters is on that kind of thing.

Reply:
> you simply blew away the pkg-config support for other platforms here which I don't find a very good idea.

What's the difference for the end user? You've got an package that builds on any platform and probably works faster (due to C compiler optimizations for the single source file). Why depending on an external binary via depext/pkg-config is so important for Unix users?

Reply:
`libsqlite3`'s build is highly configurable. Depending on your target platform (e.g. embedded systems) users may not agree with the way you choose to build it. `pkg-config` will point to the way the end-user sees it fit without this particular package having to care about this.

Besides same could be said for any other package in `opam` that depends on C bits.  At a certain point you have to put a limit with what goes in and what is left out from the `opam-repository`. So far the (wise) decision has been not to try to repackage the entire world but interact with existing package managers that are specialized for this (see the `depexts` mecanism).

Reply:
Ok, point taken. I'll try to use C sources conditionally only on Windows (or maybe integrate [conan.io](https://conan.io) support into `depext`)

Reply:
> Ok, point taken. I'll try to use C sources conditionally only on Windows (or maybe integrate [conan.io](https://conan.io) support into `depext`)

I think `depexts` for Windows will be a must at some point so any progress towards this would be welcome, I'm sure @dra27 may have thoughts on this.

Reply:
Thanks @dbuenzli and @db4 for this discussion.  I agree that this is probably best handled by adding OPAM support for depexts on Windows.  Downloading external dependencies at build time causes too many issues (invalid URLs, security issues, etc.).  Including external dependencies in the distribution is also impractical for a variety of reasons (external bug fixes, user requirements for specific versions, etc.).